troi/utils.py

```diff
@@ -2,12 +2,30 @@
 import inspect
 import os
 import traceback
+import sys
 
 from troi import Element, Artist, Release, Recording
 import troi.patch
 
 
-def discover_patches(patch_dir):
+def discover_patches():
+    """
+        Attempt to load patches from the installed patches dir as well as any patches directory in the current dir.
+    """
+
+    patches = discover_patches_from_dir("troi.patches.", os.path.join(os.path.dirname(__file__), "patches"))
+    local_patches = discover_patches_from_dir("patches.", "./patches", True)
+    return  {**patches, **local_patches}
+
+
+def discover_patches_from_dir(module_path, patch_dir, add_dot=False):
+    """
+        Load patches given the appropriate python module path and then file system path. 
+        If add_dot = True, add . to the sys.path and then remove it before this function exists.
+    """
+
+    if add_dot:
+        sys.path.append(".")
 
     patch_dict = {}
     for path in os.listdir(patch_dir):
```

Comment:
using `glob.glob("*.py")` means that you wouldn't have to do specific checks for the extension later here.
I think that os.listdir in python [doesn't return . and ..](https://docs.python.org/3/library/os.html#os.listdir), this isn't C 😄

Reply:
I'm skipping the change to glob -- it makes things more complicated. I'll remove the C like checks, sure...